test/index.test.ts

```diff
@@ -552,6 +622,81 @@ describe("mkdist with vue-tsc v1", () => {
       export default _default;
       "
     `);
+
+    expect(
+      await readFile(
+        resolve(rootDir, "dist/components/script-multi-block.vue"),
+        "utf8",
+      ),
+    ).toMatchInlineSnapshot(`
+      "<script>
+      import { defineComponent as _defineComponent } from "vue";
+      export default /* @__PURE__ */ _defineComponent({
+        __name: "script-multi-block",
+        props: {
+          msg: { type: String, required: true }
+        },
+        setup(__props, { expose: __expose }) {
+          __expose();
+          const __returned__ = {};
+          Object.defineProperty(__returned__, "__isScriptSetup", { enumerable: false, value: true });
+          return __returned__;
+        }
+      });
+      </script>
```

Comment:
this is a change of behaviour for the `.vue` files which hitherto we had just been stripping types from - I think it's fine but it does mean that types will be 'fixed' at runtime - e.g. typescript types will be made into JS ones and we may have to revisit if that ends up not working as well

Reply:
> https://vuejs.org/guide/components/props#prop-validation
> Vue will try its best to compile the type annotations into equivalent runtime prop declarations. 

I don't really understand this, according to the vue docs, their behavior shouldn't have changed, it's just that the new method now compile it earlier (when building the library)

Reply:
when importing from `#imports` or `#components` in a nuxt project these values might be different depending on modules or user code

Reply:
I see... It happens when the module use some types from vfs or something can be overrite.

I am optimistic about this, it only happens when...
- the component uses type-only `defineProps` to define props, and
- the type of the single prop changes when the module used by others (e.g. `defineProps<{ id: number }>()` to `defineProps<{ id: string }>()`, but not `defineProps<{ item: { id: number } }>()` to `defineProps<{ item: { id: string } }>()`)

I think this is more like an improper use, type-only props declarations was clearly pointed out that it will generate equivalent runtime declaration from static type analysis, devs should avoid relying on uncertain types to define props.

In the worst case, users will only get some annoying warnings, and it will not affect the prob mode. They have ample opportunity to fix these.